add methods order for `conan create` (#3682)
conan-io/docs: reference/commands/create.rst

  dependencies, you can control their build using the ``--build-test`` argument.

Methods execution order
-----------------------

> The ``conan create`` executes :ref:`methods <reference_conanfile_methods>` of a *conanfile.py* in the following order:

Reviewer: When this is properly ordered, note that you can use the `graphviz` directive to generate graphs on the fly, see https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html

> #. ``export()``

Reviewer 2: Missing ``init()``

#. ``export_sources()``
#. ``config_options()``
#. ``configure()``
#. ``layout()``
#. ``requirements()``
#. ``package_id()``
#. ``validate()``
#. ``validate_build()``

> #. ``build_requirements()``

Reviewer 2: This is not true. The call that calls ``requirements()``: https://github.com/conan-io/conan/blob/06fdb130171640c6191b1cddfb44119d67b326b6/conans/client/conanfile/configure.py#L45

#. ``build_id()``
#. ``system_requirements()``
#. ``source()``
#. ``generate()``

> #. ``imports()``

Reviewer 2: ``imports()`` doesn't exist anymore

  dependencies, you can control their build using the ``--build-test`` argument.

Methods execution order
-----------------------

The ``conan create`` executes :ref:`methods <reference_conanfile_methods>` of a *conanfile.py* in the following order:

#. Export recipe to the cache
    #. ``init()``
    #. ``set_name()``
    #. ``set_version()``
    #. ``export()``
    #. ``export_sources()``
#. Compute dependency graph
    #. ``ìnit()``
    #. ``config_options()``
    #. ``configure()``
    #. ``requirements()``
    #. ``build_requirements()``
#. Compute necessary packages
    #. ``validate_build()``
    #. ``validate()``
    #. ``package_id()``
    #. ``layout()``
    #. ``system_requirements()``
#. Install packages

>     #. ``build_id()``

Reviewer 2: I am missing the call to ``source()`` somewhere.

    #. ``generate()``
    #. ``build()``

>     #. ``package()``

Reviewer 2: These will not be called if the package is not being built from source

    #. ``package_info()``

After that, if you have a folder named *test_package* in your project or you call the ``conan create`` command with the
``--test-folder`` flag, the command will call the *conanfile.py* file inside the folder in the following order:

#. Compute dependency graph
    #. ``ìnit()``
    #. ``config_options()``
    #. ``configure()``
    #. ``requirements()``
    #. ``build_requirements()``
#. Compute necessary packages
    #. ``validate_build()``
    #. ``validate()``
    #. ``package_id()``
    #. ``layout()``
    #. ``system_requirements()``
#. Install packages
    #. ``build_id()``
    #. ``generate()``
    #. ``build()``
    #. ``package()``

>     #. ``package_info()``

Reviewer 2: This might not be fully accurate. I am not sure ``package()`` is called for ``test_package/conanfile`` at all.
It would also be missing the call to ``test()`` method.

> Steps ``3.1, 3.2, 3.3, 3.4`` will be skipped if the project is already installed. Typically, it should be installed

Reviewer 2: This numbers are not very clear to what they belong.

  dependencies, you can control their build using the ``--build-test`` argument.

Methods execution order
-----------------------

The ``conan create`` executes :ref:`methods <reference_conanfile_methods>` of a *conanfile.py* in the following order:

#. Export recipe to the cache
    #. ``init()``
    #. ``set_name()``
    #. ``set_version()``
    #. ``export()``
    #. ``export_sources()``
#. Compute dependency graph
    #. ``ìnit()``
    #. ``config_options()``
    #. ``configure()``
    #. ``requirements()``
    #. ``build_requirements()``
#. Compute necessary packages
    #. ``validate_build()``
    #. ``validate()``
    #. ``package_id()``
    #. ``layout()``
    #. ``system_requirements()``
#. Install packages
    #. ``source()``
    #. ``build_id()``
    #. ``generate()``
    #. ``build()``
    #. ``package()``
    #. ``package_info()``

Steps ``generate()``,  ``build()``, ``package()`` from *Install packages* step will not be called if the package

> is not being built from source.

Reviewer: ```suggestion
is not being built from sources.
```

After that, if you have a folder named *test_package* in your project or you call the ``conan create`` command with the

> ``--test-folder`` flag, the command will call the *conanfile.py* file inside the folder in the following order:

Reviewer 3: ```suggestion
``--test-folder`` flag, the command will invoke the methods of the *conanfile.py* file inside the folder in the following order:
```

#. Launch test_package
    #. (test package) ``init()``
    #. (test package) ``set_name()``
    #. (test package) ``set_version()``
#. Compute dependency graph
    #. (test package) ``config_options()``
    #. (test package) ``configure()``
    #. (test package) ``requirements()``
    #. (test package) ``build_requirements()``
    #. ``ìnit()``
    #. ``config_options()``
    #. ``configure()``
    #. ``requirements()``
    #. ``build_requirements()``
#. Compute necessary packages
    #. ``validate_build()``
    #. ``validate()``
    #. ``package_id()``
    #. ``layout()``
    #. (test package) ``validate_build()``
    #. (test package) ``validate()``
    #. (test package) ``package_id()``
    #. (test package) ``layout()``
    #. ``system_requirements()``
    #. (test package) ``system_requirements()``
#. Install packages
    #. ``build_id()``
    #. ``generate()``
    #. ``build()``
    #. ``package_info()``
#. Test the package
    #. (test package) ``build()``
    #. (test package) ``test()``

> The functions with *(test package)* belong to the *conanfile.py* in the *test_package* folder. Besides steps
> ``build_id()``, ``generate()``, ``build()`` inside the *Install packages* step, will be skipped if the project is
> already installed. Typically, it should be installed just as it was installed in the previous "install packages" step.

Reviewer: ```suggestion
The functions with *(test package)* belong to the *conanfile.py* in the *test_package* folder. The steps
``build_id()``, ``generate()``, ``build()`` inside the *Install packages* step will be skipped if the project is
already installed. Typically, it should be installed just as it was installed in the previous "install packages" step.
```
